jet.h

```diff
@@ -42,16 +123,43 @@ class JetCorrProvider : public CorrectionsBase<JetCorrProvider> {
                                     GenJet_pt, GenJet_eta, GenJet_phi, GenJet_mass, event);
         RVecLV shifted_p4(Jet_pt.size());
         for (int jet_idx= 0 ; jet_idx < Jet_pt.size(); ++jet_idx){
-            shifted_p4[jet_idx] = LorentzVectorM(result.pt(static_cast<int>(source))[jet_idx], Jet_eta[jet_idx],
-            Jet_phi[jet_idx], result.mass(static_cast<int>(source))[jet_idx]);
+            int scale_idx = GetScaleIdx(scale);
+            shifted_p4[jet_idx] = LorentzVectorM(result.pt(scale_idx)[jet_idx], Jet_eta[jet_idx],
+            Jet_phi[jet_idx], result.mass(scale_idx)[jet_idx]);
         }
         return shifted_p4;
     }
     RVecF getResolution(const RVecF& Jet_pt, const RVecF& Jet_eta, const float rho) const {
         return jetVarCalc.getResolution(Jet_pt, Jet_eta, rho);
     }
 
+    RVecLV getJesJet(const RVecF& Jet_pt, const RVecF& Jet_eta, const RVecF& Jet_phi,
+                    const RVecF& Jet_mass, const RVecF& Jet_rawFactor, const RVecF& Jet_area,
+                    const RVecI& Jet_jetId, const float rho, const RVecI& Jet_partonFlavour,
+                    std::uint32_t seed, const RVecF& GenJet_pt, const RVecF& GenJet_eta,
+                    const RVecF& GenJet_phi, const RVecF& GenJet_mass, int event,
+                    UncSource source, UncScale scale){
+        std::unique_ptr<JetVariationsCalculator> jvc ;
+        const UncScale jet_scale = sourceApplies(source)
+                                           ? scale : UncScale::Central;
+        const std::string& scale_str = getScaleStr(jet_scale);
+        JetCorrectorParameters jetParams(JesTxtFile_,getJesName(source));
+        jvc->addJESUncertainty(getJesName(source), jetParams);
```

Comment:
don't load the file for each jet!

Reply:
ok

Reply:
To create JetCorrectorParameters the source is needed.